priam/src/main/java/com/netflix/priam/aws/UpdateSecuritySettings.java

```diff
@@ -82,21 +77,25 @@ public void execute() {
                         .stream()
                         .map(i -> i.getHostIP() + "/32")
                         .collect(Collectors.toSet());
-        // Make sure a hole is opened for my instance.
-        // This accommodates the eventually consistent CassandraInstanceFactory.
-        // Remove once IPs are all private as there won't be any chance of a discrepancy anymore.
-        String myIp =
-                config.usePrivateIP() ? instanceInfo.getPrivateIP() : instanceInfo.getHostIP();
-        desiredAcl.add(myIp + "/32");
-        Set<String> aclToAdd = Sets.difference(desiredAcl, currentAcl);
-        if (!aclToAdd.isEmpty()) {
-            membership.addACL(aclToAdd, port, port);
-            firstTimeUpdated = true;
+        if (!config.skipDeletingOthersIngressRules()) {
+            Set<String> aclToRemove = Sets.difference(currentAcl, desiredAcl);
+            logger.info("ingress rules to delete: {}", Joiner.on(",").join(aclToRemove));
+            if (!aclToRemove.isEmpty()) {
+                membership.removeACL(aclToRemove, port, port);
+                firstTimeUpdated = true;
+            }
         }
-        Set<String> aclToRemove = Sets.difference(currentAcl, desiredAcl);
-        if (!aclToRemove.isEmpty()) {
-            membership.removeACL(aclToRemove, port, port);
-            firstTimeUpdated = true;
+        if (!config.skipUpdatingOthersIngressRules()) {
+            Set<String> aclToAdd = Sets.difference(desiredAcl, currentAcl);
+            logger.info("ingress rules to update: {}", Joiner.on(",").join(aclToAdd));
+            if (!aclToAdd.isEmpty()) {
+                int resultingSize = aclToAdd.size() + currentAcl.size();
+                if (resultingSize > config.getACLSizeWarnThreshold()) {
+                    logger.warn("We are trying to make too many ingress rules! {}", resultingSize);
```

Comment:
Should we also start emitting a metric for resultingSize (in addition to log line). Wondering how discoverable a log line alone would be.

Reply:
Point taken. Done.